pennylane/tensorbox/autograd_box.py

```diff
@@ -0,0 +1,61 @@
+# Copyright 2018-2020 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""This module contains the AutogradBox implementation of the TensorBox API.
+"""
+# pylint: disable=no-member
+import pennylane as qml
+from pennylane import numpy as np
+
+
+class AutogradBox(qml.tensorbox.TensorBox):
+    """Implements the :class:`~.TensorBox` API for ``pennylane.numpy`` tensors.
+
+    For more details, please refer to the :class:`~.TensorBox` documentation.
+    """
+
+    @staticmethod
+    def astensor(tensor):
+        return np.tensor(tensor)
+
+    def cast(self, dtype):
+        return AutogradBox(np.tensor(self.data, dtype=dtype))
+
+    def expand_dims(self, axis):
+        return AutogradBox(np.expand_dims(self.data, axis=axis))
+
+    @property
+    def interface(self):
+        return "autograd"
+
+    def numpy(self):
+        return self.data.numpy()
```

Comment:
wouldn't self.data be a numpy array here?

Reply:
Nope, `self.data` is the raw PennyLane NumPy tensor. While still a subclass of `ndarray`, so in most cases duck typing works fine, there are a few rare places where you need an actual `ndarray` representation.